code/upstream-integration-tests/src/androidTest/java/com/adobe/marketing/mobile/integration/UpstreamIntegrationTests.kt

```diff
@@ -0,0 +1,794 @@
+import com.adobe.marketing.mobile.util.TestHelper.getDispatchedEventsWith
+import com.adobe.marketing.mobile.util.TestHelper.setExpectationEvent
+import org.json.JSONObject
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+
+/**
+ * Performs validation on integration with the Edge Network upstream service.
+ */
+@RunWith(AndroidJUnit4::class)
+class UpstreamIntegrationTests {
+    private val realNetworkService = RealNetworkService()
+    private val edgeLocationHint: String = BuildConfig.EDGE_LOCATION_HINT
+    private val edgeEnvironment: String = BuildConfig.EDGE_ENVIRONMENT
+
+    @JvmField
+    @Rule
+    var rule: RuleChain = RuleChain
+        .outerRule(LogOnErrorRule())
+        .around(SetupCoreRule())
+        .around(RegisterMonitorExtensionRule())
+
+    @Before
+    @Throws(Exception::class)
+    fun setup() {
+        println("Environment var - Edge Network environment: $edgeEnvironment")
+        println("Environment var - Edge Network location hint: $edgeLocationHint")
+        ServiceProvider.getInstance().networkService = realNetworkService
+
+        // Set environment file ID for specific Edge Network environment
+        setMobileCoreEnvironmentFileID(edgeEnvironment)
+        val latch = CountDownLatch(1)
+        MobileCore.registerExtensions(listOf(Edge.EXTENSION, Identity.EXTENSION)) {
+            latch.countDown()
+        }
+        latch.await()
+
+        // Set Edge location hint if one is set for the test suite
+        if (edgeLocationHint.isNotEmpty()) {
+            print("Setting Edge location hint to: $edgeLocationHint")
+            Edge.setLocationHint(edgeLocationHint)
+        }
+        else {
+            print("No preset Edge location hint is being used for this test.")
+        }
+
+        resetTestExpectations()
+    }
+
+    @After
+    fun tearDown() {
+        realNetworkService.reset()
+    }
+
+    /**
+     * Tests that a standard sendEvent receives a single network response with HTTP code 200.
+     */
+    @Test
+    fun testSendEvent_receivesExpectedNetworkResponse() {
+        val interactNetworkRequest = TestableNetworkRequest(createURLWith(locationHint = edgeLocationHint), HttpMethod.POST)
+
+        realNetworkService.setExpectationForNetworkRequest(interactNetworkRequest, expectedCount = 1)
+
+        val experienceEvent = ExperienceEvent.Builder()
+            .setXdmSchema(mapOf("xdmtest" to "data"))
+            .setData(mapOf("data" to mapOf("test" to "data")))
+            .build()
+
+        Edge.sendEvent(experienceEvent) {}
+
+        realNetworkService.assertAllNetworkRequestExpectations()
+        val matchingResponse = realNetworkService.getResponseFor(interactNetworkRequest)
```

Comment:
should this be getResponse**s**For and return an array with all matches for this url? then we can also assert the number is greater than the expectation. This comment applies for all other tests the get the responses for asserts.